draft-documents/encrypted-data-vaults.md

```diff
@@ -79,79 +69,172 @@ one or more device(s) they control, and data is stored locally to these devices.
+
+Conversely, client-side encryption - as with Datashards - offers a high level of
+security and privacy, especially if metadata can be encrypted as well. Encryption
+is done at the individual data object level, usually aided by a keychain or wallet
+client, so the user has direct access to the private keys. This comes at a cost,
+however, since the significant responsibility of key management and recovery falls
+squarely onto the end user. In addition, the question of key management becomes
+more complex when data needs to be shared.
+
+Gateway-side encryption systems like Tahoe-LAFS take an approach that combines
+techniques from storage-side and client-side encryption architectures. These
+storage systems, typically encountered among multi-server clusters or some
+"encryption as a platform" cloud service providers, recognize that client-side
+key management may be too difficult for some users and use cases, and offer to
+perform encryption and decryption themselves, in a way that is transparent to
+the client application. At the same time, they aim to minimize the number of
+components (storage servers) that have access to the private decryption keys.
+As a result, the keys usually reside on "gateway" servers, which encrypt the
+data before passing it to the storage servers. The encryption/decryption is
+transparent to the client, and the data is opaque to the storage servers, which
+can be modular/pluggable as a result. Gateway-side encryption provides some
+benefits over storage-side systems, but also share the drawbacks: the gateway
+sysadmin controls the keys, not the user.
+
+### Encrypted Metadata and Documents vs Blobs
+
+> We kill people based on metadata.
+> - _General Michael Hayden, former director of the NSA and the CIA_
+
+Whether or not metadata can be (or is required to be) encrypted has implications
+for privacy, security, and usability of a system.
+
+Some systems, including Solid, NextCloud and Data Hubs, support the inclusion of
+arbitrary metadata on binary data blobs. IPFS, Datashards and Tahoe-LAFS do not.
+Solid metadata is written by clients per-resource using RDF. Data Hubs uses JWTs
+for per-object metadata, as well as JSON documents for additional metadata in
+Collections (also the clients' responsibility). NextCloud clients can add
+metadata to documents using WebDAV custom properties. None of these options for
+including metadata allow for it to be encrypted.
+
+### Access Interface and Control
+
+Whether data is accessed over a network, or on a local device, data objects tend
+to need globally unique identifiers. Interfaces for reading and writing data in
+stores, as well as the mechanisms to restrict or grant the ability to do so, vary
+between implementations.
+
+NextCloud and Solid both make use of existing Web standards. NextCloud uses
+WebDAV to allow client applications to read, write, and search
+data on the server's filesystem using a directory structure, and OCP for authentication.
```

Comment:
Let's change "OCP for authentication" to:

"and a [custom login flow](https://docs.nextcloud.com/server/16/developer_manual/client_apis/LoginFlow/) for local authentication."